### PR TITLE
 Don't check cache on each access

### PR DESCRIPTION
**Pull request by sgdc3**
Improves performance on large servers

### CODE REVIEW

File: src/main/java/com/github/jikoo/enchantableblocks/EnchantableBlocksPlugin.java [hunk 1]
```diff
@@ -22,6 +22,8 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
+
+import org.bukkit.Bukkit;
```
Comment (Jikoo) [inline src/main/java/com/github/jikoo/enchantableblocks/EnchantableBlocksPlugin.java:26]: Unused import.

File: src/main/java/com/github/jikoo/enchantableblocks/util/Cache.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ private Cache(final long retention, final BiFunction<K, Boolean, V> load,
 	 */
 	public void put(final K key, final V value) {
 		// Run lazy check to clean cache
-		this.lazyCheck();
+		//this.lazyCheck();
```
Comment (Jikoo) [inline src/main/java/com/github/jikoo/enchantableblocks/util/Cache.java:91]: If you're going to comment out a line of code, it should just be deleted - that's what version control is for. Several cases of this.

**Review by Jikoo — CHANGES_REQUESTED** [review]
Overall not a fan - would prefer 1a7971b and 7dbf7e1 to be separate from this PR as they're separate concerns.

**Comment by Jikoo** [comment]
Not sure why you'd want to move to a scheduler-based approach when there could just be a cooldown applied (i.e. in lazy check if now < next check time do nothing). Since the scheduler isn't running async we're still doing the same work on the main thread, just less frequently and with larger batch sizes. This should actually result in larger lag spikes.

I am now noticing that I hilariously managed to put 6000 in for the duration of a minute in several places instead of 60000.

**Comment by sgdc3** [comment]
The main concern of the previous approach is the huge allocation rate it causes, causing frequent GC pauses.
The task will never run with a much larger batch of data, in 5 seconds the server wont't unload many chunks.
I'll split the pr asap.

**Comment by Jikoo** [comment]
That's valid, forgot how aggressive new chunk unloading got. Looking at the queue I'm also not sure why I used a TreeMap instead of using a Queue and stopping when a peeked value isn't up for expiration beyond maybe my own laziness.

**Comment by Jikoo** [comment]
Okay, I've pushed a fix for my stupidity, actually pushed the gitignore I had locally and had never included (whoops), cherry picked in your removal of Guava, and added a cooldown of 10 seconds (which can be added to the config if it seems necessary) between lazy checks. Let me know if that resolves this.

I'm still going to be looking into improving the cache, but I think I might just need to rewrite the entire expiration system.

**Comment by sgdc3** [comment]
Looks good, thank you!

**Pull request closed without merging**